pandas/tests/indexes/multi/test_get_set.py

```diff
@@ -329,3 +329,21 @@ def test_set_levels_with_iterable():
         [expected_sizes, colors], names=["size", "color"]
     )
     tm.assert_index_equal(result, expected)
+
+
+def test_at_indexing_fails_multiindex():
+    # GH9259
+    cols = [("a_col", chr(i + 65)) for i in range(2)]
+    idx = [("a_row", chr(i + 65)) for i in range(2)]
+    df = pd.DataFrame(
+        np.linspace(1, 4, 4).reshape(2, 2),
+        index=pd.MultiIndex.from_tuples(idx),
+        columns=pd.MultiIndex.from_tuples(cols),
+    )
+    s = pd.Series(np.linspace(1, 2, 2), index=pd.MultiIndex.from_tuples(idx))
+
+    with pytest.raises(KeyError, match=r".+? indexing with .+? failing for MultiIndex"):
+        df.at["a_row", "A"]
```

Comment:
is there a similar problem for `at.__setitem__`?

Reply:
Multiindex with at.__setitem__ doesn't raise an exception and sets items at level 0:

```
In [1]: import pandas as pd 
   ...: import numpy as np 
   ...:  
   ...: cols = [("a_col", chr(i + 65)) for i in range(2)] 
   ...: idx = [("a_row", chr(i + 65)) for i in range(2)] 
   ...: df = pd.DataFrame( 
   ...:     np.linspace(1, 4, 4).reshape(2, 2), 
   ...:     index=pd.MultiIndex.from_tuples(idx), 
   ...:     columns=pd.MultiIndex.from_tuples(cols), 
   ...: ) 
   ...: display(df)
                                                                                                           
        a_col     
            A    B
a_row A   1.0  2.0
      B   3.0  4.0

In [2]: df.at["a_row", "A"] = 888.0 
   ...: display(df)
                                                                                                           
         a_col       
             A      B
a_row A  888.0  888.0
      B    3.0    4.0

In [3]: df.at["a_row", "A"] = [888.0, 999.0] 
   ...: display(df)
                                                                                                           
         a_col       
             A      B
a_row A  888.0  999.0
      B    3.0    4.0
```